hashintel/hash
# Implement loading state for pages

## 🌟 What is the purpose of this PR?

To improve the UX, this PR adds loading states for pages. We had some really basic loading states like `Loading...`, but now we have skeleton loaders to provide visual clarity & better UI.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1202787635216887/1202889717609176/f) _(internal)_

## 🔍 What does this change?

- Changes the default empty node to `loading` instead of `blank`, and renders it as a `LoadingView`. Used a custom view to render loading state as a react component via `renderPortal`
- Removes default outline of ProseMirror on selected nodes, we'll probably implement a custom one. 
  - [Asana task](https://app.asana.com/0/1201095311341924/1202896944903687/f) _(internal)_

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Open a page with lots of content in it
1. Refresh the page, check the skeleton loading visuals

## 📹 Demo


https://user-images.githubusercontent.com/39553853/187668387-33b584a0-2907-4911-b494-29bc435f4620.mov

<img width="500" alt="1" src="https://user-images.githubusercontent.com/39553853/187668727-ea4ad22c-1162-409e-b47a-e65ae653d0a3.png">
<img width="500" alt="2" src="https://user-images.githubusercontent.com/39553853/187668741-e02b2f32-c3ef-4789-a4d0-bb6e6e7549b7.png">
<img width="500" alt="3" src="https://user-images.githubusercontent.com/39553853/187668752-0b3ee864-eed8-4f2a-b3fc-182e632c5100.png">
